ports/espressif/Makefile

```diff
@@ -178,6 +175,8 @@ CFLAGS += $(INC) -Werror -Wall -std=gnu11 -Wl,--gc-sections $(BASE_CFLAGS) $(C_D
 
 ifeq ($(IDF_TARGET_ARCH),xtensa)
 CFLAGS += -mlongcalls
+else ifeq ($(IDF_TARGET_ARCH),riscv)
+CFLAGS += -march=rv32imc
```

Comment:
Also looks like you find this necessary. Do you have any idea why we have to roll back the arch? I did some searching on this in the esp-idf issues and couldn't find anything.

Reply:
The arch is not rolled back here. The `ESP32-C3`, `C2`, `H2` and even the `ULP Coproc` on `S2/S3` are based on `rv32imc`.

I missed your initial debug comment where you pointed out that the `CP8-beta.2` build was `rv32gc`.

As @jepler pointed out in https://github.com/adafruit/circuitpython/issues/7060#issuecomment-1281641332, `rv32gc` supports floating point arithmetic and `rv32imc` has to emulate it in software, with `rv32gc` this led to `Illegal Instruction` exception.

Reply:
In patch3, gcc was configured "--with-arch=rv32imc", so the default when invoking it without an explicit `-march=` setting was rv32imc.  In patch 5, it was configured "--with-arch=rv32gc", which causes gcc to emit some instructions (which ones?) not present in rv32imc.
```
jepler@bert:~/.espressif/tools$ ./riscv32-esp-elf/esp-2021r2-patch3-8.4.0/riscv32-esp-elf/bin/riscv32-esp-elf-gcc -v 2>&1 | grep -o -e '--with-arch=[^ ]*'
--with-arch=rv32imc
jepler@bert:~/.espressif/tools$ ./riscv32-esp-elf/esp-2021r2-patch5-8.4.0/riscv32-esp-elf/bin/riscv32-esp-elf-gcc -v 2>&1 | grep -o -e '--with-arch=[^ ]*'
--with-arch=rv32gc
```
esp-idf always set the `-march=rv32imc` flag for stuff built with cmake, but we never set it before.

Reply:
> esp-idf always set the `-march=rv32imc` flag for stuff built with cmake, but we never set it before.

Aha, that's the key point I was looking for. I didn't see this.

Reply:
esp-idf seems to set these flags always, maybe we should copy them:
```
+set(CMAKE_C_FLAGS "-Os -march=rv32imc -mdiv -fdata-sections -ffunction-sections"
+    CACHE STRING "C Compiler Base Flags")
```

Reply:
-mdiv in particular enables use of a division instruction

Reply:
@MicroDev1 can you set and test these?

Reply:
> he was running out of space on these builds.

That was because of the incorrect log level setting in sdkconfig which is fixed now.
Maybe we should evaluate other instances where low space played a role.

Reply:
> @MicroDev1 can you set and test these?

The additional `C_FLAGS` are already present in our config apart from `-mdiv` which seems to be specific to `ULP`.